@here/harp-olp-utils/lib/OlpDataProvider.ts

```diff
@@ -93,33 +89,43 @@ export class OlpDataProvider implements DataProvider {
      * @param abortSignal Optional AbortSignal to cancel the request.
      * @returns A promise delivering the data as an [[ArrayBufferLike]], or any object.
      */
-    getTile(tileKey: TileKey, abortSignal?: AbortSignal): Promise<ArrayBufferLike | {}> {
+    async getTile(tileKey: TileKey, abortSignal?: AbortSignal): Promise<ArrayBufferLike | {}> {
```

Comment:
why do you need async?

Why can't you just go:

 return this.m_versionLayerClient.getData( ... ).then(...).catch(...)?

Reply:
As in previous comment - i find async/await cleaner and more robust.

Example of "more robust", now this method never throws "synchronously" (previously it could). Now it returns rejected promise even if "sync-like" precondition failed.
 
In this case, this change is only to be consistent with rest of module.

Reply:
Good point about it not throwing a sync error, though you could argue that in such a case, a sync error likely highlights some precondition missing, and is therefore a programmer error, as opposed to some network issue or some cancellation.

I'm not such a fan of the try / catch block to enable the async in this case. I will add a comment below.

Reply:
> though you could argue that in such a case, a sync error likely highlights some precondition missing, and is therefore a programmer error, as opposed to some network issue or some cancellation.

It's not like this. If this is programmer error, then it should be `assert` or but in this case it's null check, so assert is not good because it's removed in production. Uh, oh ... runtimeAssert !?

There is no rule (either in our project or in general) that sync errors are preconditions and async are not. There is nothing that stops programmers to check & throw preconditions asynchronously and it's very common in every JS code base.

What is bad, is mixing sync and async that becomes source of hard-to-find bugs very fast, if for example you write something like this (assuming original getTile):

```
 // not async scope
 getTile(...).then(...).catch(...)
```

Will this catch our precondition ? No.

You need to
```
try {
 getTile(...).then(...).catch(...)
} catch(...) {
}
```
or call it from `async` function with (or without await).

Reply:
I'm of the opinion that the precondition should not be "caught", i.e. the program should throw the error and crash as it would in the first `getTile` example above. That makes it easier to spot the bug and fix it. Otherwise the application ends up in some undefined state and you have a fun game of find the rabbit trying to locate the issue.